client/landing/gutenboarding/components/header/index.tsx

```diff
@@ -137,14 +171,43 @@ const Header: FunctionComponent = () => {
 	}, [ newUser, handleCreateSite ] );
 
 	useEffect( () => {
+		// TODO: Cancellable?
+		// let cancel = () => undefined;
+
 		if ( newSite ) {
-			! isDomainFlow && resetOnboardStore();
-			const location = isDomainFlow
-				? `/checkout/${ newSite.blogid }/personal?redirect_to=%2Fgutenboarding%2Fdesign`
-				: `/block-editor/page/${ newSite.blogid }/home?is-gutenboarding`;
-			window.location.href = location;
+			if ( isDomainFlow ) {
+				// I'd rather not make my own product, but this works.
+				// lib/cart-items helpers did not perform well.
+				const domainProduct = {
+					meta: domain?.domain_name,
+					product_id: domain?.product_id,
+					extra: {
+						privacy_available: domain?.supports_privacy,
+						privacy: domain?.supports_privacy,
+						source: 'gutenboarding',
+					},
+				};
+
+				const go = async () => {
+					const cart: Cart = await wpcom.getCart( newSite.site_slug );
+					await wpcom.setCart( newSite.blogid, {
+						...cart,
+						products: [ ...cart.products, domainProduct ],
+					} );
```

Comment:
Requested review from @klimeryk based on the domains we're adding to the cart here.

This is a paid domain that comes from domain suggestions.

Reply:
Sorry for not getting to this on time - was overwhelmed with everything this week 🙇 
Did some testing and seems to be working well enough, even with ccTLDs requiring additional info during checkout 👍 
Though, there are a few cases that will need to be accommodated - for example, Google TLDs  (dev, page, app) require special messaging about SSL certificates that seems to be missing from this search interface. But I'm guessing that's for future iterations :)

Reply:
Thanks @klimeryk. Where can we find details about special cases and what's required? That's news to me, we'll need to add an issue with details so it's not overlooked.

cc @simison

Reply:
Please ping me on a P2 post (master thread?) for this project and I'll try my best to document them 🙇

Reply:
👍 ✅ pbAok1-6r-p2#comment-978